front/app/containers/Admin/formBuilder/components/FormBuilderSettings/index.tsx

```diff
@@ -46,18 +48,73 @@ const FormBuilderSettings = ({
 }: Props) => {
   const locales = useAppConfigurationLocales();
   const [currentTab, setCurrentTab] = useState<'content' | 'logic'>('content');
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [fieldIndexToDelete, setFieldIndexToDelete] = useState<number>();
   const { formatMessage } = useIntl();
-  const { watch } = useFormContext();
+  const { watch, setValue } = useFormContext();
+  const formCustomFields: IFlatCustomField[] = watch('customFields');
+
+  const closeModal = () => {
+    setShowDeleteModal(false);
+  };
+  const openModal = () => {
+    setShowDeleteModal(true);
+  };
 
   if (isNilOrError(locales)) {
     return null;
   }
 
+  const preDeletion = (fieldIndex: number) => {
```

Comment:
I'm open to a different name here

Reply:
I guess technically though if it opens the modal instead, it doesn't delete anything :thinking: The other function which the modal "delete" button calls removes relevant rules and then calls the delete function (for pages that have linked logic).

Reply:
I know, but still the intent is to delete a field. Maybe the user will not allow it, but the intent is still to delete. I think a well chosen name is clear when you read the call site:

```
<ContentSettings
  field={field}
  locales={locales}
  onClose={onClose}
  isDeleteDisabled={isDeleteDisabled}
  onDelete={deleteField}
/>
```
It reads "pressing the delete button will trigger the deletion of the field".

Reply:
That's true :) Ok, I'll switch it to "deleteField" :+1: Thanks!